cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/ApiCredentials.java

```diff
@@ -118,13 +121,14 @@ private static boolean validateApiUsersConfig(CruiseControlSpec ccSpec)  {
      * Parse String containing API credential config into map of API user entries.
      *
      * @param config API credential config file as a String.
+     * @param pattern Regex pattern used to parse Cruise Control API Users configuration
      *
      * @return map of API credential entries containing username, password, and role.
      */
-    /* test */ static Map<String, UserEntry> parseEntriesFromString(String config) {
+    /* test */ static Map<String, UserEntry> parseEntriesFromString(String config, Pattern pattern) {
```

Comment:
I see you are passing the pattern based on the type of Api service but how do you know that the logic within this method is going to work with a different pattern as well?
See for example the line where you split on `:` or `,` ... you are assuming these are the separators for any pattern?I don't think so. This way you are mixing the parsing logic: the pattern is coming from a specific Api service, but the splitting is assuming fixed separators. I think I suggested in another comment that this `parseEntriesFromString` as well as the `generateApiAuthFileAsString` should e part of the `ApiUsers` base class as abstract methods but being implemented in the specific `HashLoginServiceApiUsers` class.
This way you don't need to extract the type but just doing something like ...

```java
ApiUsers apiUsers = ccSpec.getApiUsers();
````

and then calling `apiUsers.parseEntriesFromString(...)` or `apiUsers.generateApiAuthFileAsString(...)` which will use the right implementation.
Wdyt? Or am I missing something here?